zipkin-web/src/main/scala/com/twitter/zipkin/web/Main.scala

```diff
@@ -55,8 +57,22 @@ trait ZipkinWebFactory { self: App =>
   val queryLimit = flag("zipkin.web.query.limit", 10, "Default query limit for trace results")
   val environment = flag("zipkin.web.environmentName", "", "The name of the environment Zipkin is running in")
 
-  def newQueryClient() = Thrift.newIface[ZipkinQuery.FutureIface]("ZipkinQuery=" + queryDest())
-  def newDependencySource() = Thrift.newIface[DependencySource.FutureIface]("DependencySource=" + queryDest())
+  // If a scribe host is configured, send all traces to it, otherwise disable tracing
+  val scribeHost = sys.env.get("SCRIBE_HOST")
+  val scribePort = sys.env.get("SCRIBE_PORT")
+  DefaultTracer.self = if (scribeHost.isDefined || scribePort.isDefined) {
```

Comment:
This is the best way to make sure "all tracing" gets affected. Otherwise, it is whack-a-mole where some use builders and others Thrift/Httpx.foo